### PR TITLE
vespalib::Monitor -> std::mutex and std::condition_variable

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge or @vekterli PR

### CODE REVIEW

**Review by toregge — APPROVED** [review]

File: storageframework/src/vespa/storageframework/generic/thread/thread.cpp [hunk 1]
```diff
@@ -3,30 +3,27 @@
...
 
 void
-Thread::interruptAndJoin(std::mutex &m, std::condition_variable &cv)
+Thread::interruptAndJoin(std::condition_variable &cv)
 {
     interrupt();
     {
-        std::lock_guard<std::mutex> guard(m);
         cv.notify_all();
     }
```
Comment (vekterli) [inline storageframework/src/vespa/storageframework/generic/thread/thread.cpp:25]: Consider removing now unused scope block

File: storage/src/vespa/storage/common/storagelink.h [hunk 2]
```diff
@@ -199,7 +201,7 @@ class Queue {
      * (0 = don't wait, -1 = forever)
      * @return  true or false if the queue was empty.
      */
-    bool getNext(std::shared_ptr<api::StorageMessage>& msg, int timeout);
+    bool getNext(std::shared_ptr<api::StorageMessage>& msg, vespalib::duration timeout);
```
Comment (vekterli) [inline storage/src/vespa/storage/common/storagelink.h:204]: Consider removing comment about "-1 = waiting forever", as that probably doesn't work anymore (did it ever?)

**Review by vekterli — APPROVED** [review]
:+1:

File: storageframework/src/vespa/storageframework/generic/thread/thread.cpp (hunk 1 above)
Comment (baldersheim) [reply storageframework/src/vespa/storageframework/generic/thread/thread.cpp]: Thanks, done

File: storage/src/vespa/storage/common/storagelink.h (hunk 2 above)
Comment (baldersheim) [reply storage/src/vespa/storage/common/storagelink.h:203]: Done.

**Pull request merged**